### PR TITLE
Implement primary key support,

### PR DESCRIPTION
**Pull request by panarch**
- [x] Adding `fetch_data` to `Store` trait
- [x] Expanding `ast::IndexItem`
- [x] Execution layer - select
- [x] Execution layer - insert
- [x] Execution layer - validate
- [x] Query planner
- [x] Integration tests
- [x] Planner unit tests

### CODE REVIEW

File: core/src/plan/join.rs [hunk 1]
```diff
@@ -24,34 +24,30 @@ pub fn plan(schema_map: &HashMap<String, Schema>, statement: Statement) -> State
...
-            SetExpr::Values(_) => body,
+            SetExpr::Values(_) => query.body,
         };
 
-        Query {
-            body,
-            limit,
-            offset,
-        }
+        query
```
Comment (ever0de) [inline core/src/plan/join.rs:42]:
```suggestion
    fn query(&self, outer_context: Option<Rc<Context<'a>>>, query: Query) -> Query {
        let body = match query.body {
            SetExpr::Select(select) => {
                let select = self.select(outer_context, *select);

                SetExpr::Select(Box::new(select))
            }
            SetExpr::Values(_) => query.body,
        };

        Query { body, ..query }
    }
```

Comment (panarch) [reply core/src/plan/join.rs]: Wow, I didn't know this kind of assignment working. thanks a lot!

File: core/src/plan/primary_key.rs [hunk 2]
```diff
@@ -0,0 +1,577 @@
...
+                    },
+                    join_operator: JoinOperator::Inner(JoinConstraint::None),
+                    join_executor: JoinExecutor::NestedLoop,
+                }],
+            },
+            selection: None,
+            group_by: Vec::new(),
+            having: None,
+            order_by: Vec::new(),
+        });
```
Comment (ever0de) [inline core/src/plan/primary_key.rs:424]: It looks that current_context should contain join relation tables, too

**Review by ever0de — APPROVED** [review]
This looks... very very... good.
Thanks to you, I learned a lot about PK. thank you

**Pull request merged**